helix_fhir_client_sdk/fhir_client.py

```diff
@@ -2715,3 +2734,17 @@ async def simulate_graph_async(
             logger=self._logger,
             auth_scopes=self._auth_scopes,
         )
+
+    def add_additional_request_headers(self, headers: Dict[str, str]) -> None:
```

Comment:
You are calling this function multiple times, and it is setting the same value each time.

What if we set this User-Agent value only once in the setter method so that we can avoid calling it multiple times?
Also, let's say someone adds another method in the future that sends headers in a request but forgets to call this 'add_additional_request_headers' method; then the User-Agent header will not get set. If we set it only once in the setter method, we can tackle both of these problems.